src/test/java/com/cloudbees/jenkins/support/CheckFilterTest.java

```diff
@@ -0,0 +1,412 @@
+                Enumeration<NetworkInterface> networkInterfaces = null;
+                networkInterfaces = NetworkInterface.getNetworkInterfaces();
+                if (networkInterfaces.hasMoreElements()) {
+                    NetworkInterface ni = networkInterfaces.nextElement();
+                    Enumeration<InetAddress> inetAddresses = ni.getInetAddresses();
+                    if (inetAddresses != null && inetAddresses.hasMoreElements()) {
+                        return inetAddresses.nextElement().toString();
+                    }
+                }
+            } catch (SocketException ignored) {
+                return null;
+            }
+            return null;
+        }
+
+        /**
+         * After creating all the contents and
+         */
+        private void generateFilteredWords() {
+            ContentFilter filter = SupportPlugin.getContentFilter().orElse(null);
+            for(FileToCheck file : fileSet) {
+                file.wordFiltered = (filter != null && file.fileIsFiltered) ? filter.filter(file.word) : file.word;
+            }
+        }
+
+        /**
+         * A filePattern pattern. Only * is allowed between to indicate subfolders.
+         * @param filePattern pattern of the file
+         * @param word the word that should be in the content, filtered or not.
+         * @param fileIsFiltered if this file is filtered
+         * @return an object to check the content of files in the bundle that matches the pattern.
+         */
+        private FileToCheck of(String filePattern, String word, boolean fileIsFiltered) {
+            words.add(word);
+            return new FileToCheck(filePattern, word, fileIsFiltered);
+        }
+
+        /**
+         * Look for a {@link FileToCheck} in the list and if one is found that matches the filePattern passed, check the
+         * content. The content should have the filtered word if filtered, in other case, the plain word.
+         * @param file filePattern path in the bundle
+         * @param content content of this filePattern
+         */
+        private void check(String file, String content) {
+            final int MAX_CONTENT_LENGTH = Integer.MAX_VALUE;
+            for (FileToCheck value : fileSet) {
+                // If there is a filePattern to check that matches this entry of the bundle, we check
+                if(value.match(file)) {
+                    if (content == null) {
+                        fail(String.format("Error checking the file %s because its content was null", file));
+                    } else {
+                        Assert.assertTrue(String.format("The file '%s' should have the word '%s'. File content:\n\n----------\n%s\n%s----------\n\n", file, value.wordFiltered, content.substring(0, Math.min(MAX_CONTENT_LENGTH, content.length())), content.length() > MAX_CONTENT_LENGTH ? "...\n(content cut off)\n" : ""), content.toLowerCase(Locale.ENGLISH).contains(value.wordFiltered.toLowerCase(Locale.ENGLISH)));
```

Comment:
It is useful for diagnosing bug in future changes but should we avoid printing the content of the file for security reasons? Maybe we are showing information of the CI. Although this information is also accessible by anyone doing a PR anywhere.